client/my-sites/plans-features-main/components/plan-upsell-modal/components/suggested-plan-section.tsx

```diff
@@ -0,0 +1,56 @@
+import { PlanSlug, getPlan } from '@automattic/calypso-products';
+import { formatCurrency } from '@automattic/format-currency';
+import styled from '@emotion/styled';
+import { useTranslate } from 'i18n-calypso';
+import usePlanPrices from 'calypso/my-sites/plans/hooks/use-plan-prices';
+import { useSelector } from 'calypso/state';
+import { getCurrentUserCurrencyCode } from 'calypso/state/currency-code/selectors';
+import { DomainName, StyledButton } from '.';
+
+const FreeDomainText = styled.div`
+	font-size: 14px;
+	line-height: 20px;
+	letter-spacing: -0.15px;
+	color: var( --studio-green-50 );
+	margin-top: 4px;
+`;
+
+export default function SuggestedPlanSection( {
+	paidDomainName,
+	suggestedPlanSlug,
+	onButtonClick,
+	isBusy,
+}: {
+	paidDomainName: string;
+	suggestedPlanSlug: PlanSlug;
+	onButtonClick: () => void;
+	isBusy: boolean;
+} ) {
+	const translate = useTranslate();
+	const planPrices = usePlanPrices( { planSlug: suggestedPlanSlug, returnMonthly: true } );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const planTitle = getPlan( suggestedPlanSlug )?.getTitle();
+
+	return (
+		<>
+			<DomainName>
+				<div>{ paidDomainName }</div>
+				<FreeDomainText>{ translate( 'Free for one year' ) }</FreeDomainText>
+			</DomainName>
+			<StyledButton fitContent busy={ isBusy } primary onClick={ onButtonClick }>
+				{ currencyCode &&
+					translate( 'Get %(planTitle)s - %(planPrice)s/month', {
+						comment: 'Eg: Get Personal - $4/month',
+						args: {
+							planTitle: planTitle as string,
+							planPrice: formatCurrency(
```

Comment:
`Non-blocking / nitpick`:

I think it's worth noting that we're moving towards using smallest unit currencies for display prices in the 2023 pricing page https://github.com/Automattic/wp-calypso/pull/81537/. 

The usage of smallest units in a currency ( Ex. $20.50 is represented as 2050 cents instead of 20.50 ) prevents rounding errors when doing math (floating point math is too inaccurate for prices).

Not something that needs to be addressed in this PR, but how would you feel about us documenting this in an issue?

Reply:
> we start handling price calculations in these modals

I think we shouldn't generally worry about smallest unit until a calculation is involved.